src/Helper/Customer/Utility.php

```diff
@@ -184,9 +187,40 @@ protected function setAddresses($shopgateCustomer, $magentoCustomer)
             $shopgateAddress->setZipcode($mageAddress->getPostcode());
             $shopgateAddress->setCountry($mageAddress->getCountryId());
             $shopgateAddress->setState($this->regions->getIsoStateByMagentoRegion($mageAddress));
-
+            $shopgateAddress->setCustomFields($this->getShopgateAddressCustomFields($mageAddress));
             $addresses[] = $shopgateAddress;
         }
         $shopgateCustomer->setAddresses($addresses);
     }
+
+    /**
+     * @param Address $mageAddress
+     *
+     * @return ShopgateOrderCustomField[]
+     */
+    protected function getShopgateAddressCustomFields($mageAddress)
+    {
+        $customFields = [];
+        foreach (self::ADDRESS_CUSTOM_FIELD_WHITELIST as $customFieldKey) {
+            $getter = 'get' . SimpleDataObjectConverter::snakeCaseToUpperCamelCase($customFieldKey);
+
+            if (!method_exists($mageAddress, $getter)) {
+                continue;
+            }
+
+            $fieldValue = $mageAddress->$getter();
+
+            if (empty($fieldValue)) {
+                continue;
+            }
```

Comment:
You are kind doing a bit too much here, just check if it has data, e.g
```php
if (!$mageAddress->hasData($customFieldKey)) {
 continue; 
}
$fieldValue = $this->getData($customField);
```
This way it checks for both existence of the field and if it is not null, also no snakeCase stuff needed.

Reply:
It doesn't appear that $magentoAddress has a hasData method or a getData method.
When i try to use them I get the error `undefined method Magento\Customer\Model\Data\Address::hasData()`

The IDE can't find the hasData or getData methods either. 

![image](https://user-images.githubusercontent.com/11764719/67678071-b07b8e80-f953-11e9-9cf5-5022d0c2f273.png)
![image](https://user-images.githubusercontent.com/11764719/67678077-b5d8d900-f953-11e9-9caf-7996abb8682c.png)

Any idea of what I am doing wrong?

Reply:
nothing, ur code is good. Mage is just inconsistent. They ported just setData ^_^

Reply:
Yeah, maybe they wan't to stop using magic and enforce to use class methods as they could be overwritten internally or by plugins to modify default setter/getter.